api/grpcserver/v2beta1/activation.go

```diff
@@ -240,15 +243,14 @@ func (s *ActivationService) Highest(
 	_ context.Context,
 	_ *spacemeshv2beta1.HighestRequest,
 ) (*spacemeshv2beta1.HighestResponse, error) {
-	highest, err := atxs.GetIDWithMaxHeight(s.db, types.EmptyNodeID, atxs.FilterAll)
-	if err != nil {
+	highest := s.atxdata.FindHighestHonest()
+	if highest == types.EmptyATXID {
 		return &spacemeshv2beta1.HighestResponse{
 			Activation: &spacemeshv2beta1.Activation{
 				Id: s.goldenAtx.Bytes(),
 			},
 		}, nil
 	}
-
 	atx, err := atxs.Get(s.db, highest)
 	if err != nil || atx == nil {
 		return nil, status.Error(codes.NotFound, fmt.Sprintf("atx id %v not found: %s", highest, err))
```

Comment:
NIT: if we want to further improve the performance of this endpoint we could omit the DB lookup here and instead use `s.atxdata.Get(id)` as a data source which contains all information of that ATX that is returned via this endpoint besides `NumUnits`.

Reply:
It makes sense, however the `atxsdata.ATX` structure doesn't contain all information that is currently returned by this endpoint. It's missing the VRF nonce (publish epoch _could_ be picked from the cache).

I will leave it for future enhancements.